### PR TITLE
build: update to Nx 16.5

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The monorepo uses Nx 16.1 with an angular.json

Closes #

## What is the new behavior?

The monorepo uses Nx 16.5.x with the project.json structure. Converted to project.json files using `nx g @nx/workspace:convert-to-nx-project` before running `nx migrate latest`

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: nx.json [hunk 1]
```diff
@@ -2,7 +2,6 @@
   "affected": {
     "defaultBase": "main"
   },
-  "npmScope": "ngrx",
```
Comment (timdeschryver) [inline nx.json:5]: Do we need to rename the `name` in the root `package.json` to `@ngrx/root`, or doesn't that matter (as in, if the prefix is set then we're good to go and the suffix doesn't matter)?

https://nx.dev/reference/nx-json#npm-scope

File: .eslintrc.json [hunk 2]
```diff
@@ -46,9 +46,7 @@
         ],
         "@typescript-eslint/naming-convention": "off",
         "@typescript-eslint/prefer-namespace-keyword": "error",
-        // because we're using object
         "@typescript-eslint/ban-types": "warn",
-        // because we have empty constructors for DI
```
Comment (markostanimirovic) [inline .eslintrc.json:51]: we could keep these comments

File: nx.json (hunk 1 above)
Comment (brandonroberts) [reply nx.json:5]: The suffix doesn't matter AFAIK https://github.com/nrwl/nx/blob/master/package.json#L2

**Review by markostanimirovic — APPROVED** [review]

**Review by markostanimirovic — APPROVED** [review]

**Comment by markostanimirovic** [comment]
🚀

**Pull request merged**